client/icons.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"net/http"
+)
+
+// Icon represents the icon of an installed Package
+type Icon struct {
+	Extension string
+	Content   []byte
+}
+
+// Icon returns the Icon belonging to an installed Package
+func (c *Client) Icon(pkg string) (Icon, error) {
```

Comment:
(nitpick) we use "Package(pkgID string)" in the other branch, I think I slightly prefer that (plus slightly prefer to be consistent here).